microsetta_public_api/api/tests/implementation_tests/test_metadata.py

```diff
@@ -145,3 +172,46 @@ def test_metadata_filter_sample_ids_alpha_metric_filter(self):
         exp = {'sample_ids': ['sample-1', 'sample-3']}
         obs = json.loads(response)
         self.assertDictEqual(exp, obs)
+
+    def test_metadata_filter_sample_ids_query_builder(self):
+        with patch.object(MetadataRepo, 'sample_id_matches') as mock_matches, \
+                patch('microsetta_public_api.repo._metadata_repo.MetadataRepo.'
+                      'categories', new_callable=PropertyMock) as \
+                mock_categories:
+            mock_matches.return_value = ['sample-1', 'sample-3']
+            mock_categories.return_value = ['age_cat']
+            response, code = filter_sample_ids_query_builder(
+                self.sample_querybuilder,
+            )
+        self.assertEqual(code, 200)
+        exp = {'sample_ids': ['sample-1', 'sample-3']}
+        obs = json.loads(response)
+        self.assertDictEqual(exp, obs)
+
+    def test_metadata_filter_sample_ids_query_builder_resource_filters(self):
+        with patch.object(MetadataRepo, 'sample_id_matches') as mock_matches, \
```

Comment:
I'm not necessarily advocating a change here right now, but I think the use of the mocks may be artificially increasing the complexity of the tests... why not just stage some data and issue a `client.post`?

Reply:
This is the unit test for the implementation layer, which is designed to test the code in `microsetta_public_api.metadata` in isolation. As such, I want to test the code without dependencies on Flask, or or other dependencies.

See the integration tests below for tests that use staged data and test the whole system end-to-end.

Reply:
Should tests for `microsetta_public_api.metadata` be under that module?

Reply:
I am not sure which tests and which module you are referring to. But generally I will admit the current testing directories structure is a bit wonky.

See #36 for details on the current test structure and a proposed re-organization of the tests in the submodules of the `microsetta_public_api.api` module.

Reply:
Made a fix above that I think should add more clarity.